modules/runner/src/main/java/org/apache/ignite/app/IgniteCliRunner.java

```diff
@@ -17,38 +17,149 @@
 
 package org.apache.ignite.app;
 
-import static picocli.CommandLine.Model.CommandSpec;
-import static picocli.CommandLine.Model.OptionSpec;
-import static picocli.CommandLine.Model.PositionalParamSpec;
-
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import com.typesafe.config.ConfigParseOptions;
+import com.typesafe.config.ConfigRenderOptions;
 import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
 import org.apache.ignite.Ignite;
 import org.apache.ignite.IgnitionManager;
+import org.apache.ignite.internal.app.EnvironmentDefaultValueProvider;
+import org.apache.ignite.network.NetworkAddress;
 import picocli.CommandLine;
-import picocli.CommandLine.Model.ArgGroupSpec;
+import picocli.CommandLine.ArgGroup;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.TypeConversionException;
 
 /**
  * The main entry point for run new Ignite node from CLI toolchain.
  */
-public class IgniteCliRunner {
+@SuppressWarnings("FieldMayBeFinal")
+@Command(name = "runner")
+public class IgniteCliRunner implements Callable<CompletableFuture<Ignite>> {
```

Comment:
I would like to ask you to add the java doc that explains how exactly parameters are overridden each and what rules are applied. For example, `--node-name` will override the env variable `IGNITE_NODE_NAME`.